src/pydata_sphinx_theme/theme/pydata_sphinx_theme/layout.html

```diff
@@ -45,7 +45,7 @@
 {% block body_tag %}
   {# set up with scrollspy to update the toc as we scroll #}
   {# ref: https://getbootstrap.com/docs/4.0/components/scrollspy/ #}
-  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="180" data-default-mode="{{ default_mode }}">
+  <body data-bs-spy="scroll" data-bs-target="#bd-toc-nav" data-offset="180" data-default-mode="{{ default_mode }}">
```

Comment:
the scrollspy in the secondary sidebar seems to not work correctly on this PR. For example on this page https://pydata-sphinx-theme--1029.org.readthedocs.build/en/1029/user_guide/web-components.html#cards on main it correctly highlights "cards" but on this PR it highlights "dropdowns" even when you click "cards" in the sidebar.

Similar problem on other pages too.  It seems like on `main` the scrollspy changes when a heading reaches ~15% down from the top of the window, but on this PR it changes when a heading reaches ~75% down from the top of the window.

Reply:
it seems to be related to https://getbootstrap.com/docs/5.2/components/scrollspy/#options. They changed the whole structure of the scrollspy to match mozilla's `IntersectionObserver.rootMargin`. I fail to understand what to change to get the previous behavior.

Reply:
if I'm reading the docs correctly, we should do something like this:

```suggestion
  <body data-bs-spy="scroll" data-bs-target="#bd-toc-nav" data-bs-root-margin="0px 0px -60%">
```

Default is `"0px 0px -25%"` which I think explains why nav items become "active" at about 75% down the page (I think the -25% explains it). But to be honest when I make the change in the browser console it seems to have no effect. This SO post suggests maybe you have to specify what "root" is? https://stackoverflow.com/a/58625634  But that doesn't seem right because I think we want to keep the default root (the whole viewport). 🤷🏻 maybe changes to `data-bs-root-margin` in the browser console aren't picked up in real time?

Reply:
I tested your solution locally and pushed it, I think it was just not working from the console. thanks